app/src/main/java/org/mozilla/fenix/tabstray/SecureTabsTrayBinding.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.tabstray
+
+import androidx.fragment.app.Fragment
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import mozilla.components.lib.state.helpers.AbstractBinding
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifAnyChanged
+import org.mozilla.fenix.ext.removeSecure
+import org.mozilla.fenix.ext.secure
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * Sets TabsTrayFragment flags to secure when private tabs list is selected.
+ */
+@OptIn(ExperimentalCoroutinesApi::class)
+class SecureTabsTrayBinding(
+    store: TabsTrayStore,
+    private val settings: Settings,
+    private val fragment: Fragment
+) : AbstractBinding<TabsTrayState>(store) {
+
+    override suspend fun onState(flow: Flow<TabsTrayState>) {
+        flow.map { it }
+            .ifAnyChanged { state ->
+                arrayOf(
+                    state.selectedPage
+                )
+            }
```

Comment:
Missed this earlier, but for reference since it's not worth sending back for now. You can write this as:
```suggestion
        flow.map { it.selectedPage }
            .ifChanged()
```
```kotlin
```